src/integration-test/java/io/aiven/kafka/connect/gcs/AbstractIntegrationTest.java

```diff
@@ -66,11 +86,21 @@ class AbstractIntegrationTest {
 
     protected static File pluginDir; // NOPMD mutable static state
     protected static String gcsEndpoint; // NOPMD mutable static state
+
+    private static final String FAKE_GCS_SERVER_VERSION = System.getProperty("fake-gcs-server-version", "latest");
     @Container
     private static final GenericContainer<?> FAKE_GCS_CONTAINER = new FixedHostPortGenericContainer(
-            "fsouza/fake-gcs-server:latest").withFixedExposedPort(GCS_PORT, GCS_PORT)
+            String.format("fsouza/fake-gcs-server:%s", FAKE_GCS_SERVER_VERSION))
+            .withFixedExposedPort(GCS_PORT, GCS_PORT)
             .withCommand("-port", Integer.toString(GCS_PORT), "-scheme", "http")
             .withReuse(true);
+    @Container
+    protected static final KafkaContainer KAFKA = new KafkaContainer("5.2.1")
```

Comment:
Tests are refactored to use single shared Kafka to avoid the network creation for each test. Testcontainers does not prune the network when container is stopped and this can cause IPv4 address pool exhaustion in Docker.